sql/parser/ast.go

```diff
@@ -620,11 +653,28 @@ func (node BytesVal) String() string {
 	return string(encodeSQLBytes(scratch[0:0], []byte(node)))
 }
 
+// IntVal represents an integer.
+type IntVal string
```

Comment:
`NumVal` being a `string` comes from the vitess code. I imagine they did that because a) vitess never interprets the values and b) to ensure that the output precisely matches the input. Switching to `type NumVal float64` has precision problems when presenting decimal (fixed point) numbers. Looks like I could use `big.Rat` for `NumVal`, but I'm hesitant to do that.

`IntVal` has none of these problems and switching this to `type IntVal uint64` probably makes sense. Let me see if it is problematic.